internal/controller/grafanaorganization_controller.go

```diff
@@ -85,44 +83,112 @@ func (r *GrafanaOrganizationReconciler) Reconcile(ctx context.Context, req ctrl.
 func (r GrafanaOrganizationReconciler) reconcileCreate(ctx context.Context, grafanaOrganization *v1alpha1.GrafanaOrganization) (ctrl.Result, error) { // nolint:unparam
 	logger := log.FromContext(ctx)
 
-	originalGrafanaOrganization := grafanaOrganization.DeepCopy()
-	// If the grafanaOrganization doesn't have our finalizer, add it.
-	if controllerutil.AddFinalizer(grafanaOrganization, v1alpha1.GrafanaOrganizationFinalizer) {
-		logger.Info("Add finalizer to Grafana Organization")
-		// Register the finalizer immediately to avoid orphaning resources on delete
-		if err := r.Client.Patch(ctx, grafanaOrganization, client.MergeFrom(originalGrafanaOrganization)); err != nil {
+	// Add finalizer first if not set to avoid the race condition between init and delete.
+	// Note: Finalizers in general can only be added when the deletionTimestamp is not set.
+	if !controllerutil.ContainsFinalizer(grafanaOrganization, v1alpha1.GrafanaOrganizationFinalizer) {
+		// We use a patch rather than an update to avoid conflicts when multiple controllers are adding their finalizer to the ClusterCR
```

Comment:
The finalizers are set to the org CR, right? I don't think we have any business with cluster CR here?
```suggestion
		// We use a patch rather than an update to avoid conflicts when multiple controllers are adding their finalizer to the organizationCR
```